src/entities/Tag.ts

```diff
@@ -71,24 +72,31 @@ export class ClientTag {
 
   /** Returns this tag and all of its sub-tags ordered depth-first */
   @action getSubTree(): Generator<ClientTag> {
-    function* tree(tag: ClientTag): Generator<ClientTag> {
+    function* tree(tag: ClientTag, iteration: number): Generator<ClientTag> {
```

Comment:
I would rename the variable iteration to depth as it not the same. In some experimental branch I added a similar check using iterations where the max value was the number of nodes in the tree. This would require some changes though and I am already struggling with refactoring the stores 😅

Reply:
Ah, sure, will rename that later. Yeah that would be better than an arbitrary limit, but it also adds more complexity. This was just a fail-safe in case a cycle in the tag structure occurs due to a bug, so that users don't have to reset their DB to a back-up